lighthouse-core/audits/resource-summary.js

```diff
@@ -0,0 +1,107 @@
+  description: 'To set budgets for the quantity and size of page resources,' +
+    ' add a budget.json file.',
+  /** [ICU Syntax] Label for an audit identifying the number of requests and kilobytes used to load the page. */
+  displayValue: `{requestCount, plural, =1 {1 request} other {# requests}}` +
+    ` • { byteCount, number, bytes } KB`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class ResourceSummary extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'resource-summary',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      requiredArtifacts: ['devtoolsLogs', 'URL'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const mainResource = await MainResource.request({devtoolsLog, URL: artifacts.URL}, context);
+    const summary = ComputedResourceSummary.summarize(networkRecords, mainResource.url);
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      {key: 'label', itemType: 'text', text: 'Resource Type'},
+      {key: 'count', itemType: 'numeric', text: 'Requests'},
+      {key: 'sizpi18e', itemType: 'bytes', text: 'Transfer Size'},
+    ];
+
+
+    /** @type {Record<LH.Budget.ResourceType,string>} */
+    const strMappings = {
+      'total': str_(i18n.UIStrings.totalResourceType),
+      'document': str_(i18n.UIStrings.documentResourceType),
+      'script': str_(i18n.UIStrings.scriptResourceType),
+      'stylesheet': str_(i18n.UIStrings.stylesheetResourceType),
+      'image': str_(i18n.UIStrings.imageResourceType),
+      'media': str_(i18n.UIStrings.mediaResourceType),
+      'font': str_(i18n.UIStrings.fontResourceType),
+      'other': str_(i18n.UIStrings.otherResourceType),
+      'third-party': str_(i18n.UIStrings.thirdPartyResourceType),
+    };
+
+    const types = /** @type {Array<LH.Budget.ResourceType>} */ (Object.keys(summary));
+    const tableContents = types.map(type => {
+      return {
+        // ResourceType is included as an "id" for ease of use.
+        // It does not appear directly in the table.
+        resourceType: type,
+        label: strMappings[type],
+        count: summary[type].count,
+        size: summary[type].size,
+      };
+    }).sort((a, b) => {
+      // Sorts table rows to be:
+      // 1st row: Total
+      // 2nd to n-1 row: Sorted by descending size
+      // Last row: Third-party
+      if (a.resourceType === 'third-party') return 1;
```

Comment:
🚲 🏠 I'm not sure if I'm the only one who finds these sort comparators very confusing to reason about for these non-numeric cases 😆 

but if I'm not alone WDYT about something like


```js
const rows = types.map(...)
const thirdPartyEntry = rows.find(r => r.resourceType === 'third-party')
const everyoneElse = rows.filter(r => r !== thirdPartyEntry
const sortedRows = [...row.sort((b, a) => b.size - a.size), thirdPartyEntry]
```

or even just

```js
// Force third-party to be last, descending by size otherwise
const sortValueA = a.resourceType === 'third-party' ? -Number.MAX_VALUE : a.size
const sortValueB = b.resourceType === 'third-party' ? -Number.MAX_VALUE : b.size 
return sortValueB - sortValueA
```